src/modules/auth/auth.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@nestjs/common';
+import * as bcrypt from 'bcrypt';
+import { JwtService } from '@nestjs/jwt';
+import { AccountsService } from '../accounts/accounts.service';
+@Injectable()
+export class AuthService {
+    constructor(private readonly accountsService: AccountsService, private readonly jwtService: JwtService,) { }
+
+    async validateAccount(email: string, pass: string) {
+        // find if user exist with this email
+        const account = await this.accountsService.findOneByEmail(email);
+        if (!account) {
+            return null;
+        }
+
+        // find if user password match
+        const match = await this.comparePassword(pass, account.password);
+        if (!match) {
+            return null;
+        }
+
+        // tslint:disable-next-line: no-string-literal
+        const { password, ...result } = account['dataValues'];
+        return result;
+    }
+
+    public async login(account) {
+        const token = await this.generateToken(account);
+        return { account, token };
+    }
+
+    public async create(account) {
+        const pass = await this.hashPassword(account.password);
```

Comment:
lets do this at model level in getter and setters.

the password column needs to have a setter that will take in a string and encrypt and store.
similarly the getter will take in the encrypted string and decrypt and return the decrypted value